packages/plugins-core/src/types/actualPlugin.ts

```diff
@@ -0,0 +1,521 @@
+> & {
+  registerSlotContent: (
+    location: SlotLocations,
+    element: ReactElement,
+  ) => string;
+  pushModal: (element: ReactElement, modalProps?: BasicModalProps) => void;
+  registerRoute: (path: string, routeElement: ReactElement) => string;
+
+  // Dashboard widget registration - wrapped for JSX elements
+  registerDashboardWidget: (
+    widgetType: string,
+    displayName: string,
+    element: ReactElement,
+    options?: {
+      defaultWidth?: number;
+      defaultHeight?: number;
+      minWidth?: number;
+      minHeight?: number;
+    },
+  ) => string;
+
+  // Theme methods - simple and direct
+  addTheme: (
+    themeId: string,
+    displayName: string,
+    colorOverrides: ThemeColorOverrides,
+    options?: {
+      baseTheme?: 'light' | 'dark' | 'midnight';
+      description?: string;
+    },
+  ) => void;
+
+  overrideTheme: (
+    themeId: 'light' | 'dark' | 'midnight' | string,
+    colorOverrides: ThemeColorOverrides,
+  ) => void;
+
+  db?: PluginDatabase;
+  q: QueryBuilder;
+
+  // Report and spreadsheet utilities
+  createSpreadsheet: () => PluginSpreadsheet;
+
+  makeFilters: (
+    conditions: Array<PluginFilterCondition>,
+  ) => Promise<PluginFilterResult>;
+};
+
+export interface ActualPlugin {
+  name: string;
+  version: string;
+  uninstall: () => void;
+  migrations?: () => PluginMigration[];
+  activate: (context: PluginContext) => void;
+}
+
+export type ActualPluginInitialized = Omit<ActualPlugin, 'activate'> & {
+  initialized: true;
+  activate: (context: HostContext & { db: PluginDatabase }) => void;
+};
+
+export interface ContextEvent {
+  payees: { payees: PayeeEntity[] };
+  categories: { categories: CategoryEntity[]; groups: CategoryGroupEntity[] };
+  accounts: { accounts: AccountEntity[] };
+  schedules: { schedules: ScheduleEntity[] };
+}
+
+export interface HostContext {
+  navigate: (routePath: string) => void;
+
+  pushModal: (
+    parameter: (container: HTMLDivElement) => void | (() => void),
+    modalProps?: BasicModalProps,
+  ) => void;
+  popModal: () => void;
+
+  registerRoute: (
+    path: string,
+    routeElement: (container: HTMLDivElement) => void | (() => void),
+  ) => string;
+  unregisterRoute: (id: string) => void;
+
+  registerSlotContent: (
+    location: SlotLocations,
+    parameter: (container: HTMLDivElement) => void | (() => void),
+  ) => string;
+  unregisterSlotContent: (id: string) => void;
+
+  on: <K extends keyof ContextEvent>(
+    eventType: K,
+    callback: (data: ContextEvent[K]) => void,
+  ) => void;
+
+  // Dashboard widget methods
+  registerDashboardWidget: (
+    widgetType: string,
+    displayName: string,
+    renderWidget: (container: HTMLDivElement) => void | (() => void),
+    options?: {
+      defaultWidth?: number;
+      defaultHeight?: number;
+      minWidth?: number;
+      minHeight?: number;
+    },
+  ) => string;
+  unregisterDashboardWidget: (id: string) => void;
+
+  // Theme methods
+  addTheme: (
```

Comment:
To keep consistent with the wording on other functions
```suggestion
  registerTheme: (
```